flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -406,7 +427,10 @@ void createIcebergTable(ObjectPath tablePath, ResolvedCatalogTable table, boolea
     for (Map.Entry<String, String> entry : table.getOptions().entrySet()) {
       if ("location".equalsIgnoreCase(entry.getKey())) {
         location = entry.getValue();
-      } else {
+      } // Catalog props are added to support CREATE TABLE LIKE in getTable().
```

Comment:
nit: put the comment lines inside the `else if` block. 

maybe even extract a `boolean isReservedKey(String key)` method? so sth like this might be a little easier to read. `location` should also be considered a reserved key.
```
if (!isReservedKey(entry.getKey())) {
    properties.put(...);
} else {
     // handling reserved keys
     if ("location".equalsIgnoreCase(entry.getKey())) {
        location = entry.getValue();
      
}